src/lib/circuit_value.ts

```diff
@@ -533,3 +546,38 @@ Circuit.constraintSystem = function <T>(f: () => T) {
   );
   return result;
 };
+
+let memoizationContext = Context.create<{
```

Comment:
new global context that is used to remember prover values which are created during `createTransaction`, to be reused in the prover for the same method. In snarkyjs, only used for the `blindingValue` right now, but I figured it would be nice to also expose that capability to the outside world -- that's what the `memoizeWitness` function below is for.